content/iOS/Practices/iOS-practices.md

```diff
@@ -0,0 +1,118 @@
+
+### Organize code specific to your feature
+
+In the root of the project, keep folders with each feature name, some examples we find in most projects are: Authentication, Profile, Home, Navigation, Networking, etc. 
+
+Within these folders, group files tightly related to the feature in question. I like to divide them into different folders as follows: Models, Views, View Models, Managers, and Protocols. 
+
+### Organize common code
+
+In the root, besides the feature-specific folders I like to keep an Extension, Helpers, Common, and Resources folders.
+
+In the Extension folder: keep extensions of the most used native controls and objects. 
+
+In the Helpers folder: keep Constant files and classes you use regularly on the whole codebase. 
+
+In Resources folder: keep any resource file your app will consume, these may be: Property lists, JSONs, Images, etc.
+
+In Common folder: keep classes reused all along the codebase including: views, managers, protocols, as usual separated by type in different folders. 
+
+So the next time I want to do an update, by a glance to the root I know the features my app implements and I can easily navigate to the file I want to update.  
+
+### Files Structure
+
+Within file classes keep related functions together in sections organized with Marks. Some examples for controllers are: View Lifecycle, Layout, User Interaction, Protocol Implementation, etc. 
+
+It is also a nice practice to separate private functions into a private extension, so that a client at a glance at the class can differentiate the interface from the implementation.  
+
+## Architecture Patterns
+
+Be very mindful of the pattern you are gonna follow in your new project and stick with it! 
+
+There's no right answer on what pattern to follow, it depends a lot on your project. I won't be enumerating every option as there are so many, and new ones keep being invented every day according to the new challenges we face as time passes, and we start seeing flaws in the existent ones.
+So, let's stick to the more common ones for now. 
+
+### MVC
+
+Like many others, at first I started coding in MVC, one of the simpler ones. 
+This pattern divides responsibilities into three class types: Models, Views, and Controllers. 
+Models deals with data logic, Views display the UI and the Controllers handles communication between the two, navigation, and business logic.  
+
+The problem with this pattern is that a lot of the responsibilities fall in the controllers' job, making them huge and ugly beasts that can be a real pain to maintain, test, and reuse in the future. 
+
+### MVVM
+
+This pattern divides responsibilities within Models, Views, and View Models, and is one of the most populars.
+Business logic and communication between Models and Views are now responsibility of the ViewModel leaving the Controller responsibility to only handle user interaction, displaying the digested information the View Model provides in Views, and handling navigation. 
```

Comment:
I think it is just `popular` without the `s`. 

`are now responsibility of the ViewModel, leaving`. I will add a comma there, otherwise, the ViewModel is leaving because the Models and Views kick him out.